pkg/kubelet/cpumanager/cpu_manager.go

```diff
@@ -216,3 +220,34 @@ func findContainerIDByName(status *v1.PodStatus, name string) (string, error) {
 	}
 	return "", fmt.Errorf("[cpumanager] unable to find ID for container with name %v in pod status (it may not be running)", name)
 }
+
+// Admit rejects a pod if its not safe to admit for node stability.
+func (m *manager) Admit(attrs *lifecycle.PodAdmitAttributes) lifecycle.PodAdmitResult {
+	if !m.policy.IsUnderPressure() {
+		return lifecycle.PodAdmitResult{Admit: true}
+	}
+
+	admit := true
+	containers := attrs.Pod.Spec.Containers
+	for _, c := range containers {
+		if c.Resources.Requests.Cpu() == nil {
+			admit = false
+			break
+		}
+	}
+	if admit {
+		return lifecycle.PodAdmitResult{Admit: true}
+	}
+
+	// reject pods with no CPU requests when under CPU pressure
+	glog.Warningf("Failed to admit pod without CPU resource request %s - node is experiencing CPU pressure", attrs.Pod)
+	return lifecycle.PodAdmitResult{
+		Admit:   false,
+		Reason:  "Evicted",
```

Comment:
Yeah, I'll look through the code for precedent.  This is what the eviction code does and this would also be the `reason` for any pods that will be evicted (not yet implemented) by shared pool depletion.